src/main/java/org/shanerx/tradeshop/listeners/ShopTradeListener.java

```diff
@@ -0,0 +1,411 @@
+			} else {
+                costName = cost.getType().toString();
+			}
+
+			if (!containsAtLeast(playerInventory, product)) {
+				buyer.sendMessage(Message.INSUFFICIENT_ITEMS.getPrefixed()
+						.replace("{ITEM}", productName.toLowerCase()).replace("{AMOUNT}", String.valueOf(amountProd)));
+				return;
+			}
+
+			if (!containsAtLeast(chestInventory, cost)) {
+				buyer.sendMessage(Message.SHOP_EMPTY.getPrefixed()
+						.replace("{ITEM}", costName.toLowerCase()).replace("{AMOUNT}", String.valueOf(amountCost)));
+				return;
+			}
+
+			if (!canExchange(chestInventory, cost, product)) {
+				buyer.sendMessage(Message.SHOP_FULL.getPrefixed()
+						.replace("{ITEM}", costName.toLowerCase()).replace("{AMOUNT}", String.valueOf(cost)));
+				return;
+			}
+
+			if (!canExchange(playerInventory, product, cost)) {
+				buyer.sendMessage(Message.PLAYER_FULL.getPrefixed()
+						.replace("{ITEM}", productName.toLowerCase()).replace("{AMOUNT}", String.valueOf(amountProd)));
+				return;
+			}
+
+			int count = amountProd, removed;
+			while (count > 0) {
+				boolean resetItem = false;
+				ItemStack temp = playerInventory.getItem(playerInventory.first(product.getType())),
+						dupitm1 = product.clone();
+				if (count > product.getMaxStackSize()) {
+					removed = product.getMaxStackSize();
+				} else {
+					removed = count;
+				}
+
+				if (removed > temp.getAmount()) {
+					removed = temp.getAmount();
+				}
+
+				product.setAmount(removed);
+				if (!product.hasItemMeta() && temp.hasItemMeta()) {
+					product.setItemMeta(temp.getItemMeta());
+					product.setData(temp.getData());
+					resetItem = true;
+				}
+
+				playerInventory.removeItem(product);
+				chestInventory.addItem(product);
+
+				if (resetItem) {
+					product = dupitm1;
+				}
+
+				count -= removed;
+			}
+
+			count = amountCost;
+			while (count > 0) {
+				boolean resetItem = false;
+				ItemStack temp = chestInventory.getItem(chestInventory.first(cost.getType())),
+						dupitm1 = cost.clone();
+				if (count > cost.getMaxStackSize())
+					removed = cost.getMaxStackSize();
+				else
+					removed = count;
+
+				if (removed > temp.getAmount())
+					removed = temp.getAmount();
+
+				cost.setAmount(removed);
+				if (!cost.hasItemMeta() && temp.hasItemMeta()) {
+					cost.setItemMeta(temp.getItemMeta());
+					cost.setData(temp.getData());
+					resetItem = true;
+				}
+
+				chestInventory.removeItem(cost);
+				playerInventory.addItem(cost);
+
+				if (resetItem) {
+					cost = dupitm1;
+				}
+
+				count -= removed;
+			}
+
+            buyer.sendMessage(Message.ON_TRADE.getPrefixed()
+                    .replace("{AMOUNT2}", String.valueOf(amountCost))
+                    .replace("{AMOUNT1}", String.valueOf(amountProd))
+                    .replace("{ITEM2}", costName.toLowerCase())
+                    .replace("{ITEM1}", productName.toLowerCase())
+                    .replace("{SELLER}", shop.getOwner().getPlayer().getName()));
+		}
+	}
+}
```

Comment:
same as above ^^

Reply:
For the most part no, just all new when the files got merged